File: services/shop-backend/src/main/java/io/miragon/shop/application/service/article/ImportArticleService.java
package io.miragon.shop.application.service.article;

import io.miragon.shop.application.port.inbound.ImportArticleUseCase;
import io.miragon.shop.application.port.outbound.ArticleRepository;
import io.miragon.shop.domain.article.Article;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class ImportArticleService implements ImportArticleUseCase {

    private static final Logger log = LoggerFactory.getLogger(ImportArticleService.class);

    private final ArticleRepository articleRepository;

    public ImportArticleService(ArticleRepository articleRepository) {
        this.articleRepository = articleRepository;
    }

    @Override
    public void importArticle(Article article) {
        articleRepository.save(article);
        log.info("Imported article {} with name '{}' and price {}", article.getId(), article.getName(), article.getPrice());
    }
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/application/service/article/ImportArticleService.kt
package io.miragon.shop.application.service.article

import io.github.oshai.kotlinlogging.KotlinLogging
import io.miragon.shop.application.port.inbound.ImportArticleUseCase
import io.miragon.shop.application.port.outbound.ArticleRepository
import io.miragon.shop.domain.article.Article
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
@Transactional
class ImportArticleService(
    private val articleRepository: ArticleRepository
) : ImportArticleUseCase {

    private val log = KotlinLogging.logger {}

    override fun importArticle(article: Article) {
        articleRepository.save(article)
        log.info { "Imported article ${article.id} with name '${article.name}' and price ${article.price}" }
    }
} 